synapse/federation/transaction_queue.py

```diff
@@ -226,7 +226,10 @@ def handle_room_events(events):
                     events_by_room.setdefault(event.room_id, []).append(event)
 
                 yield logcontext.make_deferred_yieldable(defer.gatherResults(
-                    [handle_room_events(evs) for evs in events_by_room.itervalues()],
+                    [
+                        logcontext.preserve_fn(handle_room_events)(evs)
```

Comment:
let's not `preserve_fn` for every event.. Either preserve_fn once, or use `run_in_background` (?)

Reply:
So do `logcontext.run_in_background(handle_room_events(evs))`? Isn't that equivalent to `preserve_fn`?

Reply:
```
logcontext.run_in_background(handle_room_events, evs)
```

And yes it is, except that it doesn't create a new function object on each loop.